### ISSUE TITLE
Option to forbid trailing whitespace in multiline template strings 

### ISSUE THREAD

**Issue by martinmacko47** [eslint/eslint#10502]
**What rule do you want to change?**

no-trailing-spaces

**Does this change cause the rule to produce more or fewer warnings?**

more (with enabled option)

**How will the change be implemented? (New option, new default behavior, etc.)?**

Add boolean option `ignoreTemplateStrings` (with default `true` to keep the current behaviour)

**Please provide some example code that this change will affect:**

Note: trailing whitespace is represented with `.` to be visible in the example:
```js
let str = `
${a}
.....
${b}
`
```

**What does the rule currently do for this code?**

Currently the rule does not report the trailing space.

**What will the rule do after it's changed?**

The rule with option `ignoreTemplateStrings: false` will report the trailing space.

Reasoning: There are some editors that automaticaly strip trailing space. My goal is to prevent code that would change it's meaning after using such editors as trailing space is significant in template strings. 

Ignoring template strings was added in #5786. However it would be better to be able to opt in/out for it.

**Comment by ilyavolodin** [comment]
I'm not sure this is a good idea, since trailing spaces inside templates might actually be meaningful. Templates should be preserved exactly, and if some editors strip out trailing spaces in them, I think that's a bug in the editor. Consider the following code:
```
let output = ``;
for (let a = 0; a<10; a++) {
    output += `${a}    `;
}
console.log(output);
```
Striping out trailing spaces in this (granted - bad) example would change the output.

**Comment by platinumazure** [comment]
@ilyavolodin I agree that it's probably not a great choice for default behavior and that it probably shouldn't be autofixed, but is there any harm in adding the option from an end user point of view?

**Comment by martinmacko47** [comment]
@ilyavolodin I don't see any trailing whitespace in your example. I didn't mean whitespace at the end of a string template. But at the end of source code line.

But yes, whitespace is significant in template strings. That's the reason I prefer writing code that has **no** trailing whitespace to ensure it won't disappear by a mistake while handling the file.

My goal is not to strip away the trailing whitespace by autofix, but to be able to spot smelly code that should be refactored.

**Comment by martinmacko47** [comment]
@ilyavolodin @platinumazure So are you willing to accept a PR with opt-in option `ignoreTemplateStrings`?
